jvm.c

```diff
@@ -777,6 +814,69 @@ stack_entry_t *execute(method_t *method,
             pc += 3;
             break;
 
+        /* create new object */
+        case i_new: {
+            uint8_t param1 = code_buf[pc + 1], param2 = code_buf[pc + 2];
+            uint16_t index = ((param1 << 8) | param2);
+
+            char *class_name = find_class_name_from_index(index, clazz);
+            class_file_t *target_class =
+                find_or_add_class_to_heap(class_name, prefix);
+            assert(target_class && "Failed to load class in new");
+
+            object_t *object = create_object(target_class);
+            push_ref(op_stack, object);
+
+            pc += 3;
+            break;
+        }
+
+        /* Invoke object constructor method */
+        case i_invokespecial: {
```

Comment:
How can you ensure the correctness of opcode `invokespecial`? I meant, the implementation is quite complex, and we shall introduce an effective approach to validate it. Can you provide more tests accordingly?

Reply:
I add some nested constructor tests in `Constructor.java`, and other parts in `i_invokespecial` are same as  `i_invokestatic`. Also I followed the rule of Java 11, which didn't use `invokespecial` to handle private methods anymore[[ref]](https://openjdk.org/jeps/181), so I just only tested constructors.

Reply:
Got it. Please add some comments regarding the behavior of this constructor implementation.